app/src/Pages/Portal/Contributions/Utils/ContributionsFields.js

```diff
@@ -401,59 +459,93 @@ export const validate = values => {
     typeOfContributor,
   } = values;
   const error = {};
+  const visible = {};
 
-  if (paymentMethod === 'Check' && !checkNoEmptyString(checkNumber)) {
-    error.checkNumber = 'Check number is required.';
-  }
-  if (paymentMethod === 'Money Order' && !checkNoEmptyString(checkNumber)) {
-    error.checkNumber = 'Money Order number is required.';
-  }
+  // Make these areas visible
+  visible.isPerson = !!(
+    typeOfContributor === ContributorTypeEnum.INDIVIDUAL ||
+    typeOfContributor === ContributorTypeEnum.FAMILY
+  );
 
-  const isPerson = !!individualContributorValues.includes(typeOfContributor);
-  if (
-    inKindContributionValues.includes(subTypeOfContribution) &&
-    !checkNoEmptyString(inKindType)
-  ) {
+  visible.checkSelected = !!(
+    values.paymentMethod === 'Check' || values.paymentMethod === 'Money Order'
+  );
+  visible.paymentMethod = visible.checkSelected;
+  visible.emptyOccupationLetterDate = values.occupationLetterDate === '';
+  visible.showEmployerSection =
+    values.occupation === 'Employed' && visible.isPerson;
+  visible.showInKindFields = !!inKindContributionValues.includes(
+    values.subTypeOfContribution
+  );
+  visible.showEmployerSection = inKindContributionValues.includes(
+    subTypeOfContribution
+  );
+
+  // These fields are conditionally required
+  if (visible.showEmployerSection && isEmpty(inKindType))
     error.inKindType = 'Inkind type is required';
+
+  if (visible.checkSelected && isEmpty(checkNumber)) {
+    error.checkNumber =
+      values.paymentMethod === 'Check'
+        ? 'Check number is required.'
+        : 'Money Order number is required.';
   }
+
   if (
-    subTypeOfContribution === ContributionSubTypeFieldEnum.CASH_CONTRIBUTION &&
-    !checkNoEmptyString(subTypeOfContribution)
+    subTypeOfContribution === ContributionSubTypeEnum.CASH &&
+    isEmpty(paymentMethod)
   ) {
     error.paymentMethod = 'Payment method type is required';
   }
 
   // If it's a person require first and last name
   // else require entity name
-  if (isPerson) {
-    if (!checkNoEmptyString(firstName)) {
+  if (visible.isPerson) {
+    if (isEmpty(firstName)) {
       error.firstName = 'First name is required.';
     }
-    if (!checkNoEmptyString(lastName)) {
+    if (isEmpty(lastName)) {
       error.lastName = 'Last name is required.';
     }
-  } else if (!checkNoEmptyString(entityName)) {
+  } else if (isEmpty(entityName)) {
     error.entityName = 'Name of entity is required';
   }
 
```

Comment:
I think if it's a person we will also want `occupation` to be required.  Also, right now when I select `Employed`, the employer section isn't coming up.

Reply:
And if `inkind` is selected, the Employer Section shows regardless of `occupation` selected. So, just a mix up in some of that logic on the cut and paste, I think.